cmd/monaco/deploy/deploy.go

```diff
@@ -242,3 +247,24 @@ func collectRequiresPlatformErrors(platformCoordinatesPerEnvironment Coordinates
 func platformEnvironment(e manifest.EnvironmentDefinition) bool {
 	return e.Auth.OAuth != nil
 }
+
+func validateAuthenticationWithProjectConfigs(projects []project.Project, loadedManifest *manifest.Manifest) error {
+	for _, p := range projects {
+		for envName, env := range p.Configs {
+			for _, conf := range env {
+				switch conf[0].Type.(type) {
+				case config.ClassicApiType,
+					config.SettingsType:
+					if loadedManifest.Environments[envName].Auth.Token == nil && conf[0].Skip == false {
+						return fmt.Errorf("API: %s requires token", conf[0].Type)
+					}
+				default:
+					if loadedManifest.Environments[envName].Auth.OAuth == nil && conf[0].Skip == false {
+						return fmt.Errorf("API: %s requires oAuth", conf[0].Type)
+					}
+				}
+			}
+		}
+	}
```

Comment:
I "think" you can just do nothing and continue if `Skip` is true.
If config is not skipped you can just check the credentials.... something like this:
```go
for _, p := range projects {
	for envName, env := range p.Configs {
		environmentAuth := loadedManifest.Environments[envName].Auth // Cache the environment's auth to avoid redundant lookups

		for _, conf := range env {
			if conf[0].Skip {
				continue
			}

			switch conf[0].Type.(type) {
			case config.ClassicApiType, config.SettingsType:
				if environmentAuth.Token == nil {
					return fmt.Errorf("API of type '%s' requires a token for environment '%s'", conf[0].Type, envName)
				}
			default:
				if environmentAuth.OAuth == nil {
					return fmt.Errorf("API of type '%s' requires OAuth for environment '%s'", conf[0].Type, envName)
				}
			}
		}
	}
}
```